### ISSUE TITLE
Fault surface doesnt produce any displacement[BUG]

### ISSUE THREAD

**Issue by mustaeenqazi** [Loop3D/LoopStructural#83]
Hi,

i Generated a model which has some surfaces of geological layers/formations and some faults. The faults in the model do not show any displacement. 

The fault surfaces are generated using : 
model.create_and_add_fault()

An image of the model:
![RBF+PLI_Loopstructural](https://user-images.githubusercontent.com/66418760/134264737-2ac47567-b291-4b48-9be8-ad102bf7aa78.JPG)

**Comment by mustaeenqazi** [comment]
Hi ,
Thank you very much, here is my complete code of the model and parameters,  i used fault displacement value of  400:

![model_code](https://user-images.githubusercontent.com/66418760/134651992-d3b0a0ea-c275-4fc3-beda-2d8eb0d58f0f.JPG)
![model_code2](https://user-images.githubusercontent.com/66418760/134652010-5d139dfd-7c06-4d83-bc11-ec1f9117272a.JPG)
![model_code3](https://user-images.githubusercontent.com/66418760/134652172-1d214172-9a09-4a5b-a71f-6bbfc0378442.JPG)
![model_code4](https://user-images.githubusercontent.com/66418760/134652193-5dd67102-f614-4f9a-9c01-1137407337b5.JPG)

**Comment by lachlangrose** [comment]
Hi, Can you include the model extent please and could you make a minimal working example that I can run to see the bug. It is difficult to find a bug without knowing the input parameters.

**Comment by mustaeenqazi** [comment]
Hi,
Here is a the link to a working example of my jupyter notebook: 
https://github.com/mustaeenqazi/Structural-geological-models-for-machine-learning/blob/main/perthbasin_model_from_gempy_data-PLI-creating_fault_surface_with_PLI-Copy1.ipynb

and here is the data which i used for the model : 
https://github.com/mustaeenqazi/Structural-geological-models-for-machine-learning/blob/main/perthbasin.xlsx

Thanks you.

**Comment by mustaeenqazi** [comment]
Hi thank you very much for your reply and help in understanding of loostructural 3D. 

I added the column "coord" to my data, in which I used NaN for the formations and gave 0,1,2 values for the faults. I used the same notebook as mentioned in my last comment and i got a "value error" while creating the model. this error is stated as:

ValueError: all the input arrays must have same number of dimensions, but the array at index 0 has 1 dimension(s) and the array at index 1 has 2 dimension(s)

the error doesn't occur when i remove the column "coord".

the error states that the problem is in the "model.update()" but if i remove model.update() then the same error occurs when i visualize the created model  in the next cell.

the updated data is uploaded here:
https://github.com/mustaeenqazi/Errors/blob/main/perthbasin.xlsx

and the updated notebook is here
https://github.com/mustaeenqazi/Errors/blob/main/perthbasin_model_from_gempy_data-PLI-creating_fault_surface_with_PLI.ipynb
